Projects/hCoreUI/Sources/PresentationStyle+Detented.swift

```diff
@@ -187,16 +138,15 @@ class DetentedTransitioningDelegate: NSObject, UIViewControllerTransitioningDele
                 unanimated: true
             )
 
-            Signal(after: 0.001).future
-                .onValue { [weak presentationController] _ in
-                    guard let presentationController = presentationController else { return }
-                    PresentationStyle.Detent.set(
-                        self.detents,
-                        on: presentationController,
-                        viewController: self.viewController,
-                        unanimated: true
-                    )
-                }
+            Task { @MainActor in
+                try! await Task.sleep(nanoseconds: 1_000_000_000)
```

Comment:
do we want 1 second here? and can we have  [weak presentationController, weak self]?